stubs/ruamel.yaml/ruamel/yaml/main.pyi

```diff
@@ -0,0 +1,375 @@
+from _typeshed import Unused
+from collections.abc import Callable, Iterable, Iterator, Sequence
+from pathlib import Path
+from re import Pattern
+from types import ModuleType, TracebackType
+from typing import Any, ClassVar, Final, Literal, NoReturn, Protocol, TypeVar, overload
+from typing_extensions import Self, TypeAlias, deprecated
+
+from _ruamel_yaml import CEmitter, CParser
+
+from .comments import CommentedMap, CommentedSeq
+from .compat import VersionType, _ReadStream, _WriteStream
+from .composer import Composer
+from .constructor import BaseConstructor, Constructor, RoundTripConstructor, _ConstructorFunction, _MultiConstructorFunction
+from .docinfo import DocInfo
+from .emitter import Emitter, RoundTripEmitter, _Inf, _LineBreak
+from .events import Event
+from .nodes import Node
+from .parser import Parser, RoundTripParser
+from .reader import Reader
+from .representer import BaseRepresenter, Representer, RoundTripRepresenter, _RepresenterFunction
+from .resolver import BaseResolver
+from .scanner import RoundTripScanner, Scanner
+from .serializer import Serializer
+from .tag import Tag, _TagHandleToPrefix
+from .tokens import Token, _ScalarStyle, _VersionTuple
+
+_T = TypeVar("_T")
+_Constructor = TypeVar("_Constructor", bound=BaseConstructor)
+_Representer = TypeVar("_Representer", bound=BaseRepresenter)
+
+_YAMLType: TypeAlias = str | Literal["rt", "safe", "unsafe", "full", "base"]  # noqa: Y051
+
+# type aliases to avoid name clashes and make mypy happy
+_Reader: TypeAlias = Reader
+_Scanner: TypeAlias = Scanner
+_Parser: TypeAlias = Parser
+_Composer: TypeAlias = Composer
+_Emitter: TypeAlias = Emitter
+_Serializer: TypeAlias = Serializer
```

Comment:
I think it makes more sense to use `as` imports here, e.g. `from .reader import Reader as _Reader`.

Reply:
Ok, we can keep this if you prefer.